main/tasks/asic_task.c

```diff
@@ -6,44 +6,62 @@
 
 #include "freertos/FreeRTOS.h"
 #include "freertos/task.h"
-
+#include "asic_task.h"
 #include "asic.h"
+#include "power_management_module.h"
+#include "mining_module.h"
+#include "device_config.h"
+#include "asic_task_module.h"
 
 static const char *TAG = "asic_task";
 
 // static bm_job ** active_jobs; is required to keep track of the active jobs since the
+const double NONCE_SPACE = 4294967296.0; //  2^32
+
+double ASIC_get_asic_job_frequency_ms(float frequency)
```

Comment:
Interesting to see this moved here. Kinda makes sense, at least in the current version.